src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1347,5 +1317,326 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+                Console.Write(ItemSeparator);
+
+                try
+                {
+                    // Read file content using current user - should fail, because impersonated user
+                    // isn't explicitly authorized to access the file.
+                    _ = File.ReadAllText(Environment.GetEnvironmentVariable(LocalPath));
+                    Console.Write('1');
+                }
+                catch (SecurityException)
+                {
+                    Console.Write('0');
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    Console.Write('0');
+                }
+
+                return RemoteExecutor.SuccessExitCode;
+            });
+            p.StartInfo.Environment[LocalPath] = testFilePath;
+            p.StartInfo.Environment[UncPath] = testFileUncPath;
+            p.StartInfo.StandardOutputEncoding = Encoding.UTF8;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.UseCredentialsForNetworkingOnly = true;
+
+            Action<string, string> setup = (username, _) =>
+            {
+                if (PlatformDetection.IsNotWindowsServerCore) // for this particular Windows version it fails with Attempted to perform an unauthorized operation (#46619)
+                {
+                    SetAccessControl(username, testFilePath, Path.GetDirectoryName(testFilePath), add: true);
+                }
+            };
+
+            Action<string, string> cleanup = (username, _) =>
+            {
+                if (PlatformDetection.IsNotWindowsServerCore)
+                {
+                    // remove the access
+                    SetAccessControl(username, testFilePath, Path.GetDirectoryName(testFilePath), add: false);
+                }
+            };
+
+            using var processInfo = CreateUserAndExecute(p, false, setup, cleanup);
+
+            string processUserName = Helpers.GetProcessUserName(p);
+            string output = p.StandardOutput.ReadToEnd();
+            Assert.True(p.WaitForExit(WaitInMS));
+
+            string[] splitOutput = output.Split(ItemSeparator, StringSplitOptions.None);
+            Assert.Equal(testFileContent, splitOutput[0]);
+            Assert.Equal("0", splitOutput[1]);
+            Assert.Equal(processInfo.ImpersonationAccountName?.Split('\\')?.LastOrDefault(), processUserName);
+        }
+
+        private TestProcessState CreateUserAndExecute(
+            Process process,
+            bool runImpersonated = false,
+            Action<string, string> additionalSetup = null,
+            Action<string, string> additionalCleanup = null,
+            [CallerMemberName] string memberName = "")
+        {
+            string callerIntials = new string(memberName.Where(c => char.IsUpper(c)).Take(18).ToArray());
+
+            WindowsTestAccount processAccount = new WindowsTestAccount(string.Concat("d", callerIntials));
```

Comment:
this type implements `IDisposable`, when is it going to get disposed?

Reply:
It's passed to the ctor of `TestProcessState`. That class is responsible for disposing both accounts, and the caller of `CreateUserAndExecute` is responsible for disposing it.